React Native/Navigation.md

```diff
@@ -0,0 +1,203 @@
+
+const HomeStack: React.FC<IHomeStackProps> = (props) => {
+	return (
+		<Stack.Navigator>
+			<Stack.Screen component={Home} name="Home" />
+			// ... other screens in "Home" flow
+		</Stack.Navigator>
+	);
+};
+```
+
+**IMPORTANT**: `NavigationContainer` component accepts single child component. This is not an issue if your application uses `Tab` or `Drawer` navigation. If you use only `Stacks` you need to create `RootStack` component which will then target all other `Stacks` inside separate `Stack.Screen` components.
+Using example above this `Navigator.tsx` would look like:
+
+ ```typescript
+ const RootStack = createStackNavigator();
+
+const Navigator: React.FC<INavigatorProps> = (props) => {
+	return (
+		<NavigationContainer>
+			<RootStack.Navigator>
+				<RootStack.Screen component={AuthenticationStack} name="AuthenticationStack" />
+				<RootStack.Screen component={HomeStack} name="HomeStack" />
+			</RootStack.Navigator>
+		</NavigationContainer>
+	);
+};
+```
+
+#### 1.3 TABS
+Usually, applications have one navigation tab which should be in `navigation/tabs`. When using tab navigation, each `Tab.Screen` component should target a stack with related screens.
+
+ ```typescript
+ const Tab = createTabNavigator();
+
+const TabNavigator: React.FC<ITabNavigatorProps> = (props) => {
+	return (
+		<Tab.Navigator>
+			<Tab.Screen component={AuthenticationStack} name="AuthenticationStack" />
+			<Tab.Screen component={HomeStack} name="HomeStack" />
+		</Tab.Navigator>
+	);
+};
+```
+
+#### 1.4 Drawer
+Usually, applications have one navigation tab which should be in `navigation/drawer`. When using tab navigation, each `Tab.Screen` component should target a stack with related screens.
+
+ ```typescript
+const Drawer = createDrawerNavigator();
+
+const DrawerNavigator: React.FC<IDrawerNavigatorProps> = (props) => {
+	return (
+		<Drawer.Navigator>
+			<Drawer.Screen component={AuthenticationStack} name="AuthenticationStack" />
+			<Drawer.Screen component={HomeStack} name="HomeStack" />
+		</Drawer.Navigator>
+	);
+};
+```
+
+#### 1.4 Parameters (Typescript)
+To fully use the power of Typescript, `navigation/params` folder should contain types and interfaces for all stacks used inside the application. Since React Navigation handles `Tab` and `Drawer` navigations in the background, there is no need for any type definitions.
+Each stack should have its parameters defined in a separate file. The parameters file consists of 3 main parts:
+
+ 1. List of screens used inside of stack with related `route` params defined as `type`.
+ 2. Screen `route` parameters defined as `interface`
+ 3. Stack screen props defined as `type`
+
+Example:
+ 1.
+
+ ```typescript
+export type HomeStackParamList = {
+	Home: IHomeRouteProps;
+	// ... other HomeStack screens
+};
+```
+2.
+ ```typescript
+export interface IHomeRouteProps {
+	propName: propValue;
+	...
+};
+```
+3.
+ ```typescript
+export type HomeStackScreenProps<T extends keyof HomeStackParamList = {
```

Comment:
Needs to be with closing tag:

HomeStackScreenProps<T extends keyof HomeStackParamList **>**